lib/global_id/locator.rb

```diff
@@ -106,7 +106,9 @@ def use(app, locator = nil, &locator_block)
 
       private
         def locator_for(gid)
-          @locators.fetch(normalize_app(gid.app)) { default_locator }
+          @locators.fetch(normalize_app(gid.app)) do
+            gid.respond_to?(:unscoped) ? ACTIVE_RECORD_LOCATOR : DEFAULT_LOCATOR
```

Comment:
I still think unscoped is perfectly fine to mean Active Record or not.

Maybe we shouldn’t call it an `ActiveRecordLocator` - is that what’s tripping you up?

> Den 02/06/2015 kl. 22.16 skrev Cristian Bica <notifications@github.com>:
> 
> In lib/global_id/locator.rb <https://github.com/rails/globalid/pull/73#discussion_r31565893>:
> 
> > @@ -106,7 +106,9 @@ def use(app, locator = nil, &locator_block)
> >  
> >        private
> >          def locator_for(gid)
> > -          @locators.fetch(normalize_app(gid.app)) { default_locator }
> > +          @locators.fetch(normalize_app(gid.app)) do
> > +            gid.respond_to?(:unscoped) ? ACTIVE_RECORD_LOCATOR : DEFAULT_LOCATOR
> how about descends_from_active_record? <http://api.rubyonrails.org/classes/ActiveRecord/Inheritance/ClassMethods.html#method-i-descends_from_active_record-3F>
> gid.model_class.respond_to(:descends_from_active_record?) and gid.model_class.descends_from_active_record?
> —
> Reply to this email directly or view it on GitHub <https://github.com/rails/globalid/pull/73/files#r31565893>.
> 

--
Kasper